rest/context.go

```diff
@@ -325,6 +329,30 @@ func (ctx *gorillaRequestContext) NextURL() (string, error) {
 	return u.String(), nil
 }
 
+func (ctx *gorillaRequestContext) BuildURL(routeName, resourceID string) (string, error) {
+	r, ok := ctx.Request()
+	if !ok {
+		return "", fmt.Errorf("Unable to build next url: no request")
```

Comment:
This can just be `errors.New()` since you're not using any formatting parameters. However, I'd prefer this had scope and detail.

```go
fmt.Errorf(`go-rest: unable to build the next url for route %q, resource %q`,
        routeName, resourceID)
```